html/php-pre-challenge3/index.php

```diff
@@ -1,6 +1,41 @@
 <?php
 $limit = $_GET['target'];
+if (is_numeric($limit) && $limit >= 1 && !preg_match("/^0/",$limit) && !preg_match("/[.]/",$limit)){
+} else {
+  http_response_code(400);
+  exit();
+}
 
 $dsn = 'mysql:dbname=test;host=mysql';
 $dbuser = 'test';
 $dbpassword = 'test';
+
+try {
+  $db = new PDO ($dsn , $dbuser , $dbpassword);
+} catch (PDOException $e) {
+  http_response_code(500);
+  exit();
+}
+
+// データベースから数値を取り出す
+$numbers = $db->prepare('SELECT value FROM prechallenge3 where (value <= :limit) order by value asc');
+$numbers->bindValue(':limit', $limit, PDO::PARAM_INT);
+$numbers->execute();
+while ($number = $numbers->fetch()) {
+  $num[] = $number['value'];
+}
+for($j=0;$j < 2**8-1 ;$j++){
+  $str[$j] = str_split(sprintf('%08d',decbin($j + 1)));
+  for($k=0;$k <= 7;$k++){
```

Comment:
あー、よく見たらループの回転数を固定していたのですね！
DBから取得できるレコード数は毎回同じとは限らないので決め打ちはよくないです！！

現状limitを5とかにするとレスポンスにnullが入っちゃいますね
下記のように回転数を可変にしましょう～

```suggestion
$length = count($num);
for($j=0;$j < 2**$length-1 ;$j++){
  $str[$j] = str_split(sprintf("%0${length}d",decbin($j + 1)));
  for($k=0;$k < $length;$k++){
```

Reply:
> あー、よく見たらループの回転数を固定していたのですね！
> DBから取得できるレコード数は毎回同じとは限らないので決め打ちはよくないです！！
> 
> 現状limitを5とかにするとレスポンスにnullが入っちゃいますね
> 下記のように回転数を可変にしましょう～

$limitより大きい数字はDBから取り出す時に除外するように変更したから固定だとnullになってしまうということですね！！気をつけます！！